docs/spec_draft.md

```diff
@@ -2963,6 +2969,39 @@ elements of `a` are equal to 1, otherwise the behavior is undefined.
 
 [Back to Ops](#index-of-ops)
 
+## stablehlo.tuple
+
+### Semantics
+Groups a variadic number of tensors, tokens, or tuples in `val` and produces a
+`result` tuple.
+
+### Inputs
+
+| Name  | Type                                                                |
+|-------|---------------------------------------------------------------------|
+| `val` | variadic number of tensors, tokens, or tuples of any supported type |
+
+### Outputs
+
+| Name     | Type                                                                       |
+|----------|----------------------------------------------------------------------------|
+| `result` | a variadic sized tuple of tensors, tokens, or tuples of any supported type |
+
+### Constraints
+
+  * (C1) For all i, `type(val[i])` = `type(result[i])`.
+
+### Examples
+
+```mlir
+// %var0: [1.0, 2.0]
+// %var1: {[3]}
```

Comment:
`[3]` is a constant of type `tensor<1xi32>`. A constant of type `tensor<i32>` would be written as `3` (without brackets).